netket/experimental/sampler/rules/fermion_2nd.py

```diff
@@ -77,7 +80,10 @@ def transition(rule, sampler, machine, parameters, state, key, σ):
 
         keys = jnp.asarray(jax.random.split(key, n_chains))
 
-        def _update_sample(key, σ, hoppable_clusters):
+        # we use shard_map to avoid the all-gather coming from the batched jnp.take / indexing
+        @partial(sharding_decorator, sharded_args_tree=(True, True, True))
+        @jax.vmap
```

Comment:
shouldn't this be `("key", True, True)` ? 

Also, below at the end of the file there is `_compute_hoppable_clusters_mask`which we are not sharding explicitly. I guess it passes the test so ok... but I'm mentioning it just to be sure

Reply:
Ah I see now. Can you comment this in the docstring of `sharding_decorator`?